File: javascript/bob/bob.js
var Bob = function () { };

Bob.prototype.hey = function (input) {
  const i = input;

  if (!i.trim().length)
    return "Fine. Be that way!";

  if (i.toUpperCase() === i && /[A-Z]/.test(i))
    return "Whoa, chill out!";

  if (i.slice(-1) === '?')
    return "Sure.";

  return "Whatever.";

};

module.exports = Bob;

File: csharp/bob/Bob.cs
public static class Bob
{
    public static string Hey(string statement)
    {
        string sentence = statement.Trim();
        string yell = sentence.ToUpper();
        string normal = sentence.ToLower();

        if (sentence == "")
        {
            return "Fine. Be that way!";
        }
        if (sentence == yell && sentence != normal)
        {
            return "Whoa, chill out!";
        }
        if (sentence.Substring(sentence.Length-1, 1) == "?")
        {
            return "Sure.";
        }

        return "Whatever.";
    }
}
